### ISSUE TITLE
SchnorrProof could keep the publicKey internally

### ISSUE THREAD

**Issue by JohnLCaron** [danwallach/electionguard-kotlin-multiplatform#62]
In order to validate the **SchnorrProof**, we need the publicKey (Kij in the spec), which we serialize in the GuardianRecord as **coefficient_commitments**:

```
message GuardianRecord {
  string guardian_id = 1; // guardian id
  uint32 x_coordinate = 2;    // x coordinate in the secret polynomial
  ElementModP guardian_public_key = 3;
  repeated ElementModP coefficient_commitments = 4;  // Kij, j=0...quorum-1
  repeated SchnorrProof coefficient_proofs = 5;
}

message SchnorrProof {
  ElementModQ challenge = 3; // c in the spec
  ElementModQ response = 4; // u in the spec
}
```

could be:

```
message GuardianRecord {
  string guardian_id = 1;       // guardian id
  uint32 x_coordinate = 2;    // x coordinate in the secret polynomial
  repeated SchnorrProof coefficient_proofs = 5;
}

message SchnorrProof {
  ElementModP public_key = 1; // k in the spec
  ElementModQ challenge = 3; // c in the spec
  ElementModQ response = 4; // u in the spec
}
```

A bit better since its clear there must be a 1-1 match of proof and  public_key
The **guardian_public_key** is also not needed since its == k0.

Also open to better naming.

**Comment by JohnLCaron** [comment]
Spec 1.51 has this (p 40):

4.5.1 Schnorr proofs
Guardian records contain Schnorr proofs of knowledge of the secret polynomial coefficients.

| information     | key        | value | symbol |
|----------|-------------|-------|----|
|Public element  |    "public key" |BigInt4096 |K_i,j|
|Challenge value  |   "challenge" | BigInt256  | c_i,j
|Responses value    | "response" |BigInt256  | v_i,j
|Usage description  | "usage"   |  string| |

So, publicKey should be part of the SchnorrProof record. I dont see any point to the "usage" field.

**Comment by danwallach** [comment]
Yeah. I'm completely okay with removing text fields that convey nothing useful.

**Comment by JohnLCaron** [comment]
implemented in PR #158
